internal/uploader/events_uploader.go

```diff
@@ -0,0 +1,264 @@
+
+	var formBuffer bytes.Buffer
+	w := multipart.NewWriter(&formBuffer)
+	fw, err := w.CreatePart(mimeHeader)
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "failed creating multipart section for request body")
+	}
+
+	if _, err := fw.Write(buffer.Bytes()); err != nil {
+		return nil, "", err
+	}
+
+	if err := w.Close(); err != nil {
+		return nil, "", err
+	}
+	return &formBuffer, w.FormDataContentType(), nil
+}
+
+func sendRequest(req *http.Request) error {
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return errors.Wrapf(err, "failed to send request to %s", req.URL)
+	}
+	if res != nil {
+		log.Infof("Response from %s: %v", req.URL, res.Status)
+		if res.StatusCode > 199 && res.StatusCode < 300 {
+			log.Infof("Request ID: %+v", res.Header.Get("X-Rh-Insights-Request-Id"))
+		}
+		body, err := ioutil.ReadAll(res.Body)
+		if err == nil {
+			log.Debugf("Response body %s", string(body))
+		}
+	}
+	return nil
+}
+
+func prepareFiles(ctx context.Context, db *gorm.DB, cluster *common.Cluster, eventsHandler eventsapi.Handler, pullSecret *validations.PullSecretCreds) (*bytes.Buffer, error) {
+	buffer := &bytes.Buffer{}
+	gz := gzip.NewWriter(buffer)
+	tw := tar.NewWriter(gz)
+	filesCreated := 0
+
+	// errors creating files below will be ignored since failing to create one of the files
+	// doesn't mean the rest of the function should fail
+	if err := clusterFile(tw, cluster, pullSecret); err != nil {
+		log.WithError(err).Warnf("failed finding cluster events with cluster ID %s", cluster.ID)
+	} else {
+		filesCreated++
+	}
+
+	infraEnvID, err := hostsFile(db, tw, cluster)
+	if err != nil {
+		log.WithError(err).Warnf("failed finding host(s) events associated with cluster ID %s", cluster.ID)
+	} else {
+		filesCreated++
+	}
+
+	if err := infraEnvFile(db, tw, infraEnvID, cluster.ID); err != nil {
+		log.WithError(err).Warnf("failed finding infra-env events with ID %s", infraEnvID)
+	} else {
+		filesCreated++
+	}
+
+	if err := eventsFile(ctx, cluster.ID, eventsHandler, tw); err != nil {
+		log.WithError(err).Warnf("failed finding events associated with cluster ID %s", cluster.ID)
+	} else {
+		filesCreated++
+	}
+
+	// There are no files to upload
+	if filesCreated == 0 {
+		return nil, errors.Errorf("no event data files created for cluster %s", cluster.ID)
+	}
+
+	// produce tar
+	if err := tw.Close(); err != nil {
+		return nil, errors.Wrap(err, "failed closing tar file")
+	}
+	// produce gzip
+	if err := gz.Close(); err != nil {
+		return nil, errors.Wrap(err, "failed closing gzip file")
+	}
+	return buffer, nil
+}
+
+func eventsFile(ctx context.Context, clusterID *strfmt.UUID, eventsHandler eventsapi.Handler, tw *tar.Writer) error {
+	if eventsHandler == nil {
+		return errors.Errorf("failed to get events for cluster %s, events handler is nil", clusterID)
+	}
+	dbEvents, err := eventsHandler.V2GetEvents(ctx, clusterID, nil, nil, models.EventCategoryMetrics, models.EventCategoryUser)
+	if err != nil {
```

Comment:
> This query however does not bring the host events which I thing you want.

Is this considered a host event? 
```
 {
  "category": "user",
  "cluster_id": "20001bc6-2135-45c7-a475-a61e4023a750",
  "event_time": "2023-02-07T17:18:00.623Z",
  "host_id": "e2a4e446-90bb-4c8a-8be1-04459f6a1be5",
  "infra_env_id": "6d946cba-1238-482b-89a1-397b95b86c64",
  "message": "Host e2a4e446-90bb-4c8a-8be1-04459f6a1be5: Successfully registered",
  "name": "host_registration_succeeded",
  "request_id": "ac57d64f-b12a-4344-b0da-9c576a5ab827",
  "severity": "info"
 },
```

Reply:
> Since the events here are limited to a cluster I am not sure we will have tons of them but still it is worth having some number here. 

Sorry, I'm not sure what you mean by some number here?

Reply:
> Another problem is that you are keep sending old events from previous installation attempts. Do you have a mechanism to handle duplications down the road? Is it a concern at all?

@rccrdpccl and I have discussed data duplication a little bit, but I think it's a good topic to nail down. 

Initially, we discussed mitigating duplicate data from the receiving end. In other words, when the consumer of the ingress (kafka) server gets the data, it won't store any data that it's already seen before. 

To determine if it's already seen the data, we spoke about using the username and cluster id.

@rccrdpccl any thoughts? Would you like an additional field to be added to determine duplicate data?

Reply:
1. Yes. The example you pasted here is a host level event
2. By numbers I mean how many events we have per cluster of 5 node, 20 nodes. We can extract this data by querying our prod db - I can show you how.
3. For identification you will also need the host_id and infraenv_id (for unbound events) and a timestamp. On multiple installations of the same cluster we have the same events (logically speaking) but their timestamp differ,